apps/server/src/modules/deletion/domain/service/deletion-request.service.ts

```diff
@@ -48,14 +46,22 @@ export class DeletionRequestService {
 		return deletionRequest;
 	}
 
-	async findAllItemsToExecute(limit: number, failed = false): Promise<DeletionRequest[]> {
-		const deletionRequests = failed
-			? await this.deletionRequestRepo.findAllFailedItems(limit, this.olderThan, this.newerThan)
+	public async findAllItemsToExecute(limit: number, getFailed = false): Promise<DeletionRequest[]> {
+		const newerThan = new Date(Date.now() - this.thresholdNewer);
+		const olderThan = new Date(Date.now() - this.thresholdOlder);
+		const deletionRequests = getFailed
+			? await this.deletionRequestRepo.findAllFailedItems(limit, olderThan, newerThan)
 			: await this.deletionRequestRepo.findAllItems(limit);
 
 		return deletionRequests;
 	}
 
+	public async findInProgressCount(): Promise<number> {
+		const newerThan = new Date(Date.now() - this.thresholdOlder);
```

Comment:
is it correct that this newerThan is differntly calculated to the one ion line 50 ?

Reply:
yes, this is correct
the login is, in case for findAllItemsToExecute, we look for all records between a time interval. Basically, the failed / pending older than 5 minutes but not older than 4 days

for the findInProgressCount, we need to exclude those requests which are right now being processed. So, that will mean exclude all pending which are not older than 5 min.
(After 5 min, they are considered dead, and will be picked up by the other cron job).